### PR TITLE
Spark chart

### PR DESCRIPTION
**Pull request by rporres**
Adapted from https://github.com/kubernetes/charts/tree/master/stable/spark to our needs

* Moved worker to DaemonSet to match HDFS deployment
* Removed zeppelin deployment
* Added NodeSelectors
* Moved to ClusterIPs as we're not in a cloud provider
* Adapted to 2.2.0
* Added spark ui proxy integration

Working on https://github.com/src-d/backlog/issues/831

### CODE REVIEW

**Comment by bzz** [comment]
Awesome! Let me get back to it after running Berserker Spark jobs on our cluster tomorrow!

**Comment by smeruelo** [comment]
Every chart I've seen so far has separated files for deployments and services (skeleton generated by `helm create` is this way too). Is there a reason for having both things together in the same file here? Maybe we could split the deployment files.

File: spark/README.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+## Prerequisites
+
+* Assumes that serviceAccount tokens are available under hostname metadata. (Works on GKE by default) URL -- http://metadata/computeMetadata/v1/instance/service-accounts/default/token
+
+## Installing the Chart
+
+To install the chart with the release name `my-release`:
+
+```bash
+$ helm install --name my-release stable/spark
```
Comment (smeruelo) [inline spark/README.md]: The path should be just "spark", right? There's no "stable" directory in this repository.

File: spark/templates/NOTES.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+1. Get the Spark URL to visit by running these commands in the same shell:
+  
+  NOTE: It may take a few minutes for the LoadBalancer IP to be available.
+  You can watch the status of by running 'kubectl get svc --namespace {{ .Release.Namespace }} -w {{ template "webui-fullname" . }}'
```
Comment (rporres) [inline spark/templates/NOTES.txt]: will change

File: spark/README.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+* http://spark.apache.org/
+
+Adapted from https://github.com/kubernetes/charts/tree/master/stable/spark
+
+## Chart Details
+This chart will do the following:
+
+* 1 x Spark Master with port 8080 exposed using a ClusterIP service
+* A DaemonSet for Spark workers that can be bound to certain nodes using
```
Comment (rporres) [inline spark/README.md]: ...a node selector. Will change

File: spark/README.md (hunk 1 above)
Comment (rporres) [inline spark/README.md]: you're right!

**Comment by rporres** [comment]
I adapted the chart from the official repo that had the services and deployments together. Since we have four services, I think it is clearer to have them together with the deployments they belong to...

**Comment by rporres** [comment]
Corrected as per @smeruelo suggestions

File: spark/templates/_helpers.tpl [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+{{- $name := default .Chart.Name .Values.WebUi.Name -}}
+{{- printf "%s-%s" .Release.Name $name | trunc 24 -}}
+{{- end -}}
+
+{{- define "worker-fullname" -}}
+{{- $name := default .Chart.Name .Values.Worker.Name -}}
+{{- printf "%s-%s" .Release.Name $name | trunc 24 -}}
+{{- end -}}
+
+{{- define "zeppelin-fullname" -}}
```
Comment (bzz) [inline spark/templates/_helpers.tpl]: 😢

Comment (smeruelo) [inline spark/templates/_helpers.tpl]: Sorry :(((

Comment (rporres) [inline spark/templates/_helpers.tpl]: I did it at some point. I guess when I added spark-ui-proxy

File: spark/Chart.yaml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# Inspired from official spark chart
```
Comment (bzz) [inline spark/Chart.yaml]: One thing on licensing - AFAIK original chart was Apache 2.0 licensed (which is compatible with GPLv3 in this direction), so may be a link to original one https://github.com/kubernetes/examples/tree/master/staging/spark and a bit more substantial attributions in README might be in order?

Of course that depends on how much inspiration was taken :)
Anyway, it's not copyright law, but an etiquette here.

**Comment by rporres** [comment]
@smeruelo I've arranged commits to keep them at minimum. You can merge this when you want ;-)

**Pull request merged**